CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.23)
```

Comment:
This requirement fails approx. half of the CI workflows. Do we need 3.23? Or can we go slightly lower as well?

Reply:
3.23 is when `FILE_SET` was introduced in CMake: https://cmake.org/cmake/help/latest/command/target_sources.html#file-sets
We can just bump the LCG versions, 103 is on 3.23 and 104 is on 3.26 so any of those should work